cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -790,6 +806,92 @@ public void testResources() {
+                .withCpu(brokerCapacityCpu)
+                .withOverrides()
+                    .addNewOverride()
+                        .addToBrokers(0)
+                        .withCpu(brokerCapacityOverrideCpu)
+                    .endOverride()
+                .build();
+
+        verifyBrokerCapacity(nodes, storage, resourceRequirements, brokerCapacityOne, brokerCapacityOverrideCpu, brokerCapacityCpu, brokerCapacityCpu);
+
+        // In this test case, when override is set for a broker, it takes precedence over resource limit
+        io.strimzi.api.kafka.model.balancing.BrokerCapacity brokerCapacityTwo = new BrokerCapacityBuilder()
+                .withOverrides()
+                    .addNewOverride()
+                        .addToBrokers(0)
+                        .withCpu(brokerCapacityOverrideCpu)
+                    .endOverride()
+                .build();
+
+        verifyBrokerCapacity(nodes, storage, resourceRequirements, brokerCapacityTwo, brokerCapacityOverrideCpu, resourceLimitCpu, resourceLimitCpu);
+
+        /* In this test case, when neither the override nor the CPU resource request are configured but the CPU
+           resource limit is set for the Kafka brokers, the resource limit will be used as the CPU capacity */
+        ResourceRequirements resourceRequirements3 = new ResourceRequirementsBuilder()
+                .addToLimits("cpu", new Quantity(resourceLimitCpu))
+                .build();
+
+        io.strimzi.api.kafka.model.balancing.BrokerCapacity brokerCapacityThree = new BrokerCapacityBuilder()
+                .build();
+
+        verifyBrokerCapacity(nodes, storage, resourceRequirements3, brokerCapacityThree, resourceLimitCpu, resourceLimitCpu, resourceLimitCpu);
+
+        // In this test case, when neither the override nor the CPU resource request or limits are configured, the CPU capacity will be set to DEFAULT_CPU_CORE_CAPACITY
+        ResourceRequirements resourceRequirements2 = new ResourceRequirementsBuilder()
+                .build();
+
+        verifyBrokerCapacity(nodes, storage, resourceRequirements2, brokerCapacityThree, BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY, BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY, BrokerCapacity.DEFAULT_CPU_CORE_CAPACITY);
+    }
+
+    private void verifyBrokerCapacity(Set<NodeRef> nodes,
+                                     Map<String, Storage> storage,
+                                     ResourceRequirements resourceRequirements,
+                                     io.strimzi.api.kafka.model.balancing.BrokerCapacity brokerCapacity,
+                                     String brokerOneCpuValue,
+                                     String brokerTwoCpuValue,
+                                     String brokerThreeCpuValue) {
+
+        HashMap<String, ResourceRequirements> resources = new HashMap<>();
+        resources.put("pool1", resourceRequirements);
```

Comment:
Why do we need to add the resourceRequirements to this `resources` map? I understand it has something to do with NodePools but do we need to define it for these tests? Is there anyway we can just have it defined in the Kafka spec?

Reply:
@kyguy The resources and storage are now defined at the pool level. Not at the Kafka level anymore.

Reply:
As Jakub said above